pkg/suggestion/v1beta1/pbt/service.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2022 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import grpc
+import logging
+import numpy as np
+import os
+import shutil
+import uuid
+
+from pkg.apis.manager.v1beta1.python import api_pb2
+from pkg.apis.manager.v1beta1.python import api_pb2_grpc
+from pkg.suggestion.v1beta1.internal.search_space import (
+    HyperParameter,
+    HyperParameterSearchSpace,
+)
+import pkg.suggestion.v1beta1.internal.constant as constant
+from pkg.suggestion.v1beta1.internal.trial import Trial, Assignment, Annotations
+from pkg.suggestion.v1beta1.internal.base_health_service import HealthServicer
+
+logger = logging.getLogger(__name__)
+
+
+_REQUIRED_SETTINGS = ["checkpoint_dir", "n_population", "truncation_threshold"]
```

Comment:
Since `checkpoint_dir` will be common across various algorithms, we can move it to [constant.py](https://github.com/kubeflow/katib/blob/280eaed8cc27e25241fdcd0517354ed5deb732d9/pkg/suggestion/v1beta1/internal/constant.py)